src/tqec/templates/scale.py

```diff
@@ -0,0 +1,96 @@
+import marshal
+import typing as ty
+from operator import xor
+
+from tqec.exceptions import TQECException
+
+
+def default_scaling_function(k: int) -> int:
+    return 2 * k
+
+
+def get_fixed_scaling_function(fixed_scale: int) -> ty.Callable[[int], int]:
+    return lambda _: fixed_scale
+
+
+class Dimension:
+    def __init__(
+        self,
+        initial_value: int,
+        scaling_function: ty.Callable[[int], int] | None = None,
```

Comment:
I understand that using a `scaling_function: Callable` instead of a `scaling_factor: int` can make the dimension more flexible. However, it might add some complexity to the serialization (when serialized to bytes with marshal) and communication between the frontend and backend, although I'm not very familiar with frontend development.

According to Austin's suggestion, in most cases, `2*k` should be sufficient. While there might be situations where `2*k+1` could be useful, linear scaling with `m*k + b` should cover most cases. Therefore, it seems reasonable to use a scaling factor like `m` or `(m, b)` instead of introducing the `scaling_function`.

What's your thoughts on this?

Reply:
I agree. Using a function as input is way too much freedom and makes the task of encoding this in JSON very challenging for both the backend and frontend. I agree with `(m, b)` being likely sufficient. I will try to make that change.

Reply:
Should be done. Also the `LinearFunction` type is really simple, and we can extend it without any issue to a polynomial type.